src/status_im/ui/screens/wallet/views.cljs

```diff
@@ -49,21 +49,24 @@
     :action              #(re-frame/dispatch [:navigate-to :transactions-history])}])
 
 (defn- render-asset [{:keys [symbol icon decimals amount]}]
-  [react/view
-   [list/item
-    [list/item-image icon]
-    [react/view {:style styles/asset-item-value-container}
-     [react/text {:style               styles/asset-item-value
-                  :number-of-lines     1
-                  :ellipsize-mode      :tail
-                  :accessibility-label (str (-> symbol name clojure.string/lower-case) "-asset-value-text")}
-      (wallet.utils/format-amount amount decimals)]
-     [react/text {:style           styles/asset-item-currency
-                  :uppercase?      true
-                  :number-of-lines 1}
-      (clojure.core/name symbol)]]]])
-
-
+  (let [asset-value (re-frame/subscribe [:asset-value symbol :USD])]
```

Comment:
Use defview instead

Reply:
@goranjovic I've tried what you suggested by using `defview`, but it failed to render the view. This warning shown: `Functions are not valid as a React child. This may happen if you return a Component instead of <Component /> from render. Or maybe you meant to call this function rather than return it.` Can you tell what I did wrong with these changes below?

```diff
@@ -48,8 +48,8 @@
     :icon                :icons/transaction-history
     :action              #(re-frame/dispatch [:navigate-to :transactions-history])}])
 
-(defn- render-asset [{:keys [symbol icon decimals amount]}]
-  (let [asset-value (re-frame/subscribe [:asset-value symbol :USD])]
+(views/defview render-asset [{:keys [symbol icon decimals amount]}]
+  (views/letsubs [asset-value [:asset-value symbol :USD]]
     [react/view {:style styles/asset-item-container}
      [list/item
       [list/item-image icon]
@@ -66,7 +66,7 @@
       [react/text {:style           styles/asset-item-price
                    :uppercase?      true
                    :number-of-lines 1}
-       (if @asset-value (str "$" @asset-value) "...")]]]))
+       (if asset-value (str "$" asset-value) "...")]]]))
```

Reply:
You would have to change as well `wrap-render-fn` https://github.com/status-im/status-react/blob/develop/src/status_im/ui/components/list/views.cljs#L120 to not to call `as-element` on `render-fn` when `render-fn` returns a component and not hiccup. 
Otherwise we could extract `reagent-render`.
 I would say maybe not worth doing in the scope of this PR, and move back to normal render, what do you reckon @goranjovic ?

Reply:
Yeah, let's leave it as it is and possibly add a comment for the posterity